gammapy/cube/make.py

```diff
@@ -95,21 +99,34 @@ def run(self, obs_list):
             exclusion_mask = self.maps.get('exclusion', None)
             if exclusion_mask is not None:
                 exclusion_mask, _ = exclusion_mask.make_cutout(
-                    obs.pointing_radec, 2 * self.offset_max, mode=self.cutout_mode
+                    obs.pointing_radec, 2 * self.offset_max, mode=self.cutout_mode,
                 )
 
+            # Make maps for this observation
             map_maker_obs = MapMakerObs(
                 obs=obs,
                 geom=cutout_map.geom,
                 fov_mask=fov_mask,
                 exclusion_mask=exclusion_mask,
             )
-            maps = map_maker_obs.run()
+            maps_obs = map_maker_obs.run(selection)
+
+            # Stack observation maps to total
+            if 'counts' in selection:
+                if 'counts' not in self.maps:
```

Comment:
Maybe a similar solution could be applied as below in the `__init__`:
```
for map_type in selection:
      self.maps[map_type] = Map.from_geom(self.geom)
```

Reply:
Of course this would work better with a test that all types passed belong to the required list